serving/src/main/java/feast/serving/service/HistoricalServingService.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2019 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.serving.service;
+
+import feast.serving.ServingAPIProto;
+import feast.serving.ServingAPIProto.*;
+import feast.serving.ServingAPIProto.Job.Builder;
+import feast.serving.specs.CachedSpecService;
+import feast.storage.api.retrieval.FeatureSetRequest;
+import feast.storage.api.retrieval.HistoricalRetrievalResult;
+import feast.storage.api.retrieval.HistoricalRetriever;
+import feast.storage.connectors.bigquery.retrieval.BigQueryHistoricalRetriever;
+import io.grpc.Status;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+
+public class HistoricalServingService implements ServingService {
+
+  private static final Logger log =
+      org.slf4j.LoggerFactory.getLogger(HistoricalServingService.class);
+
+  private final HistoricalRetriever retriever;
+  private final CachedSpecService specService;
+  private final JobService jobService;
+
+  public HistoricalServingService(
+      HistoricalRetriever retriever, CachedSpecService specService, JobService jobService) {
+    this.retriever = retriever;
+    this.specService = specService;
+    this.jobService = jobService;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public GetFeastServingInfoResponse getFeastServingInfo(
+      GetFeastServingInfoRequest getFeastServingInfoRequest) {
+    try {
+      BigQueryHistoricalRetriever bigQueryHistoricalRetriever =
+          (BigQueryHistoricalRetriever) retriever;
+      return GetFeastServingInfoResponse.newBuilder()
+          .setType(FeastServingType.FEAST_SERVING_TYPE_BATCH)
+          .setJobStagingLocation(bigQueryHistoricalRetriever.jobStagingLocation())
+          .build();
+    } catch (Exception e) {
+      return GetFeastServingInfoResponse.newBuilder()
+          .setType(FeastServingType.FEAST_SERVING_TYPE_BATCH)
+          .build();
+    }
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public GetOnlineFeaturesResponse getOnlineFeatures(GetOnlineFeaturesRequest getFeaturesRequest) {
+    throw Status.UNIMPLEMENTED.withDescription("Method not implemented").asRuntimeException();
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public GetBatchFeaturesResponse getBatchFeatures(GetBatchFeaturesRequest getFeaturesRequest) {
```

Comment:
I'm finding it hard to limit comments here. This is probably out of scope for the storage refactor, but having the serving service do job management here (creating a job Id and kicking off the job) seems to be an incorrect abstraction. I think this should be a part of what the job service does.

Reply:
Completely agree on both counts, the abstraction/layer boundaries and the scope of storage refactor.

I'm not sure where/when to address it, if we're going to have all this serving refactoring in the PR too.

Reply:
Well I just realized that my configuration clean up #525 is the yak shave of the century. I am trying to limit the scope of that as well, but I am touching all of the same places that @zhilingc is because configuration is heavily used by stores. I've even now rebased on top of this very PR.

I think we should try and limit the already large scope of this PR to only the most critical aspects of the storage API.

Reply:
For now, for historical stores at least, the serving service *only* manages jobs, because historical serving is done async, and historical retrieval is analogous to starting a retrieval job. I'd argue that until there is a need for the two to be defined separately (e.g. if serving does some other job management), it's fine as is.